wom/src/main/scala/wom/types/WomIntegerType.scala

```diff
@@ -12,11 +13,19 @@ case object WomIntegerType extends WomPrimitiveType {
     case i: Integer => WomInteger(i)
     case n: JsNumber if n.value.isValidInt => WomInteger(n.value.intValue())
     case i: WomInteger => i
-    case WomLong(i) if (i >= Int.MinValue && i <= Int.MaxValue) => WomInteger(i.toInt)
-    case WomLong(i) => throw new RuntimeException(s"Tried to convert a long value $i into an integer but it was outside the bounds of acceptable Integers, namely ${Int.MinValue} <-> ${Int.MaxValue}")
-    case s: WomString => WomInteger(s.value.toInt)
-    case s: String => WomInteger(s.toInt)
-    case s: JsString => WomInteger(s.value.toInt)
+    case WomLong(i) if i.inIntRange => WomInteger(i.toInt)
+    case WomLong(i) => throw new RuntimeException(
+      s"Tried to convert a Long value $i into an Int but it was outside the bounds of acceptable Ints, namely ${Int.MinValue} <-> ${Int.MaxValue}")
+    case WomOptionalValue(WomIntegerType, Some(i)) => i
```

Comment:
Could you recurse in the case of a filled optional, rather than enumerating all of the possible cases? (it should be impossible but what if we accidentally end up with a `Int??`)

Reply:
I can try